### PR TITLE
Split scenario "base state (no pending weeks)" into two sub scenarios

### PR DESCRIPTION
**Pull request by rocketnova**
## Ticket

Related to #242 

## Changes

- Splits the Base State scenarios into the current 4 known scenarios (more refinement TBD)
- Updates scenarios to be numbered
- Converts scenario type enum from an enum with string members to an enum with number members
- Adds a type alias for i18n translation strings so that we can use the type `I18nString` instead of just `string` for extra clarity

## Context

While going through the content matrix, I discovered that I didn't break out the scenarios at the correct level of granularity in order to display the content we want to show. After getting confirmation about this from @nicoleslaw and Diana, this PR corrects that.

## Testing 

`yarn test`

### CODE REVIEW

File: public/locales/en/claim-status.json [hunk 1]
```diff
@@ -3,9 +3,18 @@
...
+  "base-no-pending-active": {
+    "description": "Your claim is active and there are no pending issues at this time."
+  },
+  "base-no-pending-inactive": {
     "description": "There are no pending issues at this time."
+  },
+  "conditional-next-steps": {
+    "certify-no-pending": "<0>Keep certifying</0> as long as you need benefits.",
+    "certify-pending": "<0>Keep certifying</0> as long as you need benefits, so your payments are not delayed further.",
+    "contact-info": "Make sure we have your current phone number and email address."
```
Comment (rocketnova) [inline public/locales/en/claim-status.json:18]: Note: This PR pulls in some content that I'm not yet using in this PR.

File: utils/getScenarioContent.tsx [hunk 2]
```diff
@@ -3,14 +3,12 @@ import { Claim, ClaimDetailsContent, ClaimStatusContent, ScenarioContent } from
 export enum ScenarioType {
   PendingDetermination = 'Pending determination scenario',
   BasePending = 'Base state with pending weeks',
-  BaseNoPending = 'Base state with no pending weeks',
+  BaseNoPendingActive = 'Base state with no pending weeks; Active claim',
+  BaseNoPendingInactive = 'Base state with no pending weeks; Inactive claim',
```
Comment (lomky) [inline utils/getScenarioContent.tsx:7]: I find myself really torn on staying brief and wanting more clarity. 

Pending is overloaded and applies to both Determinations and Weeks, so should we always include the associated Noun?

Active/Inactive apply the a whole other object (the claim), should we include that?

But then we have names like:
```
  BaseNoPendingWeeksActiveClaim 
  BaseNoPendingWeeksInactiveClaim 
```
is that just getting absurd and hard to parse?

Comment (rocketnova) [reply utils/getScenarioContent.tsx:7]: Totally hear you. I'll take another pass at this.

File: stories/Page.stories.tsx [hunk 3]
```diff
@@ -12,24 +13,23 @@ export default {
   decorators: [withNextRouter],
   argTypes: {
     scenario: {
-      options: Object.keys(ScenarioType),
-      mapping: Object.keys(ScenarioType), // return the key instead of the value
```
Comment (rocketnova) [inline stories/Page.stories.tsx:16]: Note: Turns out `mapping` was not needed, so I removed it.

File: tests/utils/apiGatewayStub.test.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    expect(response.hasCertificationWeeksAvailable).toBe(false)
+  })
+
+  it('is correct for Scenario 10', () => {
+    const response = apiGatewayStub(ScenarioType.Scenario10)
+    expect([null, [], false, undefined]).toContain(response.pendingDetermination)
+    expect(response.hasPendingWeeks).toBe(true)
+    expect(response.hasCertificationWeeksAvailable).toBe(true)
+  })
+})
```
Comment (rocketnova) [inline tests/utils/apiGatewayStub.test.tsx:39]: Note: The tests for scenarios 7-10 are pretty repetitive, but I thought it would help readability to have them broken out separately. Also I think these might change soon. Let me know if it would be better to make this one test that verifies against a lookup table instead.

File: tests/utils/getScenarioContent.test.tsx [hunk 5]
```diff
@@ -1,98 +1,76 @@
...
+  it('are returned as expected', () => {
+    const baseScenarios = [
+      ScenarioType.Scenario7,
+      ScenarioType.Scenario8,
+      ScenarioType.Scenario9,
+      ScenarioType.Scenario10,
+    ]
+    for (const scenarioType of baseScenarios) {
+      expect(getScenario(apiGatewayStub(scenarioType))).toBe(scenarioType)
+    }
```
Comment (rocketnova) [inline tests/utils/getScenarioContent.test.tsx:74]: Note: This is a basically just testing that `getScenario()` can process the output for our stub, so I added tests for our stub.

File: utils/getScenarioContent.tsx [hunk 6]
```diff
@@ -1,75 +1,86 @@
...
+  Scenario9,
+  Scenario10,
+}
+
+export const ScenarioTypeNames = {
+  [ScenarioType.Scenario1]: 'Pending determination scenario',
+  [ScenarioType.Scenario7]: 'Base state; No pending weeks; No weeks to certify',
+  [ScenarioType.Scenario8]: 'Base state; No pending weeks; Has weeks to certify',
+  [ScenarioType.Scenario9]: 'Base state; Has pending weeks; No weeks to certify',
+  [ScenarioType.Scenario10]: 'Base state; Has pending weeks; Has weeks to certify',
```
Comment (rocketnova) [inline utils/getScenarioContent.tsx:16]: Note: This is the major change. After speaking with Nicole and Diana today, we agreed to call scenarios by number instead of a nick name. As part of that, I converted the enum to a numeric enum, which is able to do automatic [reverse mappings](https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings). Enums are a little finicky with Typescript, so I added the `numericEnum.tsx` utility to help.

File: utils/getScenarioContent.tsx [hunk 7]
```diff
@@ -1,75 +1,86 @@
...
-  // and hasPendingWeeks is false
-  else if (claimData.hasPendingWeeks === false) {
-    return ScenarioType.BaseNoPending
-  }
-  // This is unexpected
-  // @TODO: Log the scenario and display 500
+
+  // Otherwise display a Base State scenario.
   else {
-    // throw new Error('Unexpected scenario')
```
Comment (rocketnova) [inline utils/getScenarioContent.tsx:37]: Note: The new logic I went over with Nicole and Diana shouldn't leave an opportunity for an unknown scenario. We will also have an additional layer of API gateway response validation (#150) that will make unexpected response values not possible.

File: utils/getScenarioContent.tsx (hunk 6 above)
Comment (lomky) [reply utils/getScenarioContent.tsx:16]: I like it. Can we add a comment to the code that points to the fully-written-out Scenario descriptions?

Comment (rocketnova) [reply utils/getScenarioContent.tsx:16]: @lomky Yes for sure. But I'm not sure exactly where you are thinking we should add the comment? In `getScenarioContent.tsx`?

Comment (lomky) [reply utils/getScenarioContent.tsx:16]: I was thinking right above this const, or if we do topline comments on the files, there would also be good!

**Review by lomky — APPROVED** [review]
🎉
